### ISSUE TITLE
v2.13.1 Exit automatically after running

### ISSUE THREAD

**Issue by ttsite** [xmrig/xmrig#949]
As with the previous version, try compiling all of these problems with vs2015 vs2017 and gcc. The same is true for downloading compiled software directly from the website. The software before replacing the new algorithm will work properly.

Testing system windows Server 2008
CPU E5-2620 V3
 CPU E5-2620 v4
CPU E5-2690 v2

**Comment by ttsite** [comment]
Looking forward to your solution, today you get a new version of the test or run after the automatic exit. Replace the old algorithm software can work normally. Attempted to download the website compiled in a variety of ways is still the same. I also tried to compile the next vs2015 vs2017 GCC mode. Has been running after the automatic exit. Thank you!!!

**Comment by ttsite** [comment]
V2.13.x The original CPU default war 75% of the machine as long as it is replaced by V2.13x version, if it can work, it will immediately increase the CPU occupancy to 100%.

**Comment by vivin321** [comment]
hi
did you solve your issue ?
my miner which I compiled it exit as soon as runnng it 
how to  fix this ?

**Comment by ttsite** [comment]
> hi
> did you fix your issue ?
> my miner which I compiled it exit as soon as runnng it
> how to fix this ?

Hello, there has been no solution, so far has not found a solution, now V2.13x is not available on my machine, as long as the point to open just to connect to the mine automatically quit. Attempted to compile a variety of ways and websites to download the author compiled, all this is the case. Very anxious can not update the new version. Before March 9, there is no new version to solve my machine. In this case, all my machines can only stop working. It's very impatient.

**Comment by xmrig** [comment]
Again, please provide all information (including configs) how to reproduce this issue, preferably with original binaries, if you modify source I need this too. If I can't reproduce the issue I can't help.

**Comment by vivin321** [comment]
my configuration  is just adding the wallet and pool data to the source code
like that
in pools.cpp file :
m_data.push_back(Pool("Pool Address",Port,"WALLET_ADDRESS","PASSWOD"));
and then compile via this line

cmake .. -G "Unix Makefiles" -DXMRIG_DEPS=C:/Users/john/Desktop/xmrig-deps/gcc/x86 
make
the output exit as soon as run it

**Comment by vivin321** [comment]
her is the compile process :

toto@toto-PC MINGW32 ~
$ cd C:/xmrig/build

toto@toto-PC MINGW32 /c/xmrig/build
$ cmake .. -G "Unix Makefiles" -DXMRIG_DEPS=c:/xmrig-deps/gcc/x86
-- The C compiler identification is GNU 7.4.0
-- The CXX compiler identification is GNU 7.4.0
-- Check for working C compiler: C:/msys32/mingw32/bin/cc.exe
-- Check for working C compiler: C:/msys32/mingw32/bin/cc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/msys32/mingw32/bin/c++.exe
-- Check for working CXX compiler: C:/msys32/mingw32/bin/c++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found UV: C:/xmrig-deps/gcc/x86/lib/libuv.a
-- Found OpenSSL: C:/xmrig-deps/gcc/x86/lib/libcrypto.a (found version "1.1.1") 
-- Looking for syslog.h
-- Looking for syslog.h - not found
-- Found MHD: C:/xmrig-deps/gcc/x86/lib/libmicrohttpd.a
-- Configuring done
-- Generating done
-- Build files have been written to: C:/xmrig/build

toto@toto-PC MINGW32 /c/xmrig/build
$ make
Scanning dependencies of target cpuid
[  1%] Building C object src/3rdparty/libcpuid/CMakeFiles/cpuid.dir/cpuid_main.c.obj
[  3%] Building C object src/3rdparty/libcpuid/CMakeFiles/cpuid.dir/asm-bits.c.obj
[  4%] Building C object src/3rdparty/libcpuid/CMakeFiles/cpuid.dir/recog_amd.c.obj
[  6%] Building C object src/3rdparty/libcpuid/CMakeFiles/cpuid.dir/recog_intel.c.obj
[  7%] Building C object src/3rdparty/libcpuid/CMakeFiles/cpuid.dir/libcpuid_util.c.obj
[  9%] Linking C static library libcpuid.a
[  9%] Built target cpuid
Scanning dependencies of target xmrig
[ 10%] Building CXX object CMakeFiles/xmrig.dir/src/api/NetworkState.cpp.obj
[ 12%] Building CXX object CMakeFiles/xmrig.dir/src/App.cpp.obj
[ 13%] Building CXX object CMakeFiles/xmrig.dir/src/base/io/Json.cpp.obj
[ 15%] Building CXX object CMakeFiles/xmrig.dir/src/base/io/Watcher.cpp.obj
[ 16%] Building CXX object CMakeFiles/xmrig.dir/src/base/kernel/Entry.cpp.obj
[ 18%] Building CXX object CMakeFiles/xmrig.dir/src/base/kernel/Process.cpp.obj
[ 20%] Building CXX object CMakeFiles/xmrig.dir/src/base/kernel/Signals.cpp.obj
[ 21%] Building CXX object CMakeFiles/xmrig.dir/src/base/net/Pool.cpp.obj
[ 23%] Building CXX object CMakeFiles/xmrig.dir/src/base/net/Pools.cpp.obj
[ 24%] Building CXX object CMakeFiles/xmrig.dir/src/base/tools/Arguments.cpp.obj
[ 26%] Building CXX object CMakeFiles/xmrig.dir/src/base/tools/Handle.cpp.obj
[ 27%] Building CXX object CMakeFiles/xmrig.dir/src/base/tools/String.cpp.obj
[ 29%] Building CXX object CMakeFiles/xmrig.dir/src/common/config/CommonConfig.cpp.obj
[ 30%] Building CXX object CMakeFiles/xmrig.dir/src/common/config/ConfigLoader.cpp.obj
[ 32%] Building CXX object CMakeFiles/xmrig.dir/src/common/config/ConfigWatcher.cpp.obj
[ 33%] Building CXX object CMakeFiles/xmrig.dir/src/common/Console.cpp.obj
[ 35%] Building CXX object CMakeFiles/xmrig.dir/src/common/crypto/Algorithm.cpp.obj
[ 36%] Building CXX object CMakeFiles/xmrig.dir/src/common/crypto/keccak.cpp.obj
[ 38%] Building CXX object CMakeFiles/xmrig.dir/src/common/log/BasicLog.cpp.obj
[ 40%] Building CXX object CMakeFiles/xmrig.dir/src/common/log/ConsoleLog.cpp.obj
[ 41%] Building CXX object CMakeFiles/xmrig.dir/src/common/log/FileLog.cpp.obj
[ 43%] Building CXX object CMakeFiles/xmrig.dir/src/common/log/Log.cpp.obj
[ 44%] Building CXX object CMakeFiles/xmrig.dir/src/common/net/Client.cpp.obj
[ 46%] Building CXX object CMakeFiles/xmrig.dir/src/common/net/Job.cpp.obj
[ 47%] Building CXX object CMakeFiles/xmrig.dir/src/common/net/strategies/FailoverStrategy.cpp.obj
[ 49%] Building CXX object CMakeFiles/xmrig.dir/src/common/net/strategies/SinglePoolStrategy.cpp.obj
[ 50%] Building CXX object CMakeFiles/xmrig.dir/src/common/net/SubmitResult.cpp.obj
[ 52%] Building CXX object CMakeFiles/xmrig.dir/src/common/Platform.cpp.obj
[ 53%] Building CXX object CMakeFiles/xmrig.dir/src/core/Config.cpp.obj
[ 55%] Building CXX object CMakeFiles/xmrig.dir/src/core/Controller.cpp.obj
[ 56%] Building CXX object CMakeFiles/xmrig.dir/src/Mem.cpp.obj
[ 58%] Building CXX object CMakeFiles/xmrig.dir/src/net/Network.cpp.obj
[ 60%] Building CXX object CMakeFiles/xmrig.dir/src/net/strategies/DonateStrategy.cpp.obj
[ 61%] Building CXX object CMakeFiles/xmrig.dir/src/Summary.cpp.obj
[ 63%] Building CXX object CMakeFiles/xmrig.dir/src/workers/CpuThread.cpp.obj
[ 64%] Building CXX object CMakeFiles/xmrig.dir/src/workers/Handle.cpp.obj
[ 66%] Building CXX object CMakeFiles/xmrig.dir/src/workers/Hashrate.cpp.obj
[ 67%] Building CXX object CMakeFiles/xmrig.dir/src/workers/MultiWorker.cpp.obj
[ 69%] Building CXX object CMakeFiles/xmrig.dir/src/workers/Worker.cpp.obj
[ 70%] Building CXX object CMakeFiles/xmrig.dir/src/workers/Workers.cpp.obj
[ 72%] Building CXX object CMakeFiles/xmrig.dir/src/xmrig.cpp.obj
[ 73%] Building RC object CMakeFiles/xmrig.dir/res/app.rc.obj
[ 75%] Building CXX object CMakeFiles/xmrig.dir/src/App_win.cpp.obj
[ 76%] Building CXX object CMakeFiles/xmrig.dir/src/base/io/Json_win.cpp.obj
[ 78%] Building CXX object CMakeFiles/xmrig.dir/src/common/Platform_win.cpp.obj
[ 80%] Building CXX object CMakeFiles/xmrig.dir/src/Mem_win.cpp.obj
[ 81%] Building CXX object CMakeFiles/xmrig.dir/src/core/cpu/AdvancedCpuInfo.cpp.obj
[ 83%] Building CXX object CMakeFiles/xmrig.dir/src/core/cpu/Cpu.cpp.obj
[ 84%] Building C object CMakeFiles/xmrig.dir/src/crypto/c_groestl.c.obj
[ 86%] Building C object CMakeFiles/xmrig.dir/src/crypto/c_blake256.c.obj
[ 87%] Building C object CMakeFiles/xmrig.dir/src/crypto/c_jh.c.obj
[ 89%] Building C object CMakeFiles/xmrig.dir/src/crypto/c_skein.c.obj
[ 90%] Building CXX object CMakeFiles/xmrig.dir/src/crypto/CryptonightR_gen.cpp.obj
[ 92%] Building CXX object CMakeFiles/xmrig.dir/src/api/Api.cpp.obj
[ 93%] Building CXX object CMakeFiles/xmrig.dir/src/api/ApiRouter.cpp.obj
[ 95%] Building CXX object CMakeFiles/xmrig.dir/src/common/api/Httpd.cpp.obj
[ 96%] Building CXX object CMakeFiles/xmrig.dir/src/common/api/HttpRequest.cpp.obj
[ 98%] Building CXX object CMakeFiles/xmrig.dir/src/common/net/Tls.cpp.obj
[100%] Linking CXX executable xmrig.exe
[100%] Built target xmrig

toto@toto-PC MINGW32 /c/xmrig/build

**Comment by vivin321** [comment]
@xmrig waiting for help please

**Comment by vivin321** [comment]
after compiled the xmrig I created cmd file and wrote this code in it 
@echo off
xmrig.exe
pause

and I got this 

unable to open "C:\xmrig\build\config.json".
No valid configuration found. Exiting.
Press any key to continue . . .

**Comment by ttsite** [comment]
> Again, please provide all information (including configs) how to reproduce this issue, preferably with original binaries, if you modify source I need this too. If I can't reproduce the issue I can't help.

Every time I just want to connect to the mine address, when I see that the CPU is occupied, I automatically quit. This situation did not appear in the previous version of the new algorithm. Just now, I tested the version of v2.12.0, which is also automatic exit. Because the original version of 2.83 is stable and has not been replaced, I saw that the version of v2.13X appeared automatic exit and tested the version of v2.12x under this situation. It should start from this version, I hope the author can see from here. Look for the reasons above the version. Think about where the major changes were made at that time. Thank you!!
Test System and Processor
Windows Server 2008
CPU E5-2620 V3
CPU E5-2620 V4
CPU E5-2690 V2

**Comment by ttsite** [comment]
> after compiled the xmrig I created cmd file and wrote this code in it
> @echo off
> xmrig.exe
> pause
> 
> and I got this
> 
> unable to open "C:\xmrig\build\config.json".
> No valid configuration found. Exiting.
> Press any key to continue . . .

Hello, have you solved the problem of automatic exit? Thank you!! I've been doing this all the time.

**Comment by xmrig** [comment]
https://github.com/xmrig/xmrig/issues/957#issuecomment-468890667 bug with `m_data.push_back` also fixed.

**Comment by vivin321** [comment]
solved via xmrig-dev 
thank you

**Comment by ttsite** [comment]
> [#957 (comment)](https://github.com/xmrig/xmrig/issues/957#issuecomment-468890667) bug with `m_data.push_back` also fixed.

I found that auto-exit only happened on E5 26xx CPU. I downloaded the xmrig-dev version issued by the author yesterday. It still happened on E5 26xx level CPU, and only on 64-bit version. If I use 32-bit version on 64-bit system, I can run normally. If I use 64-bit version, it will not work. When I connect to the pool after running, it will happen. Shipping has withdrawn.
